### ISSUE TITLE
Login is not possible if you don't spell name perfectly with capitalization.

### ISSUE THREAD

**Issue by LosGlennos** [victorzki/makiato#29]
#### Example:
- LosGlennos -> passes
- losglennos -> fails
#### Proposed solution:

Have an extra field in the database called `username_lower` where we do all our queries from. This way we could do something like this:

``` javascript
var username = req.body.username.toLowerCase();
User.findOne({username_lower: username}, function(err, user) { ... });
```

**Comment by alveflo** [comment]
Good find. Don't really agree to expand the database with an additional field though... How about we either lower the username before posting the form or lower the username in the passport module;

``` javascript
User.findOne({ 'username' :  username.toLowerCase() }
```

**Comment by LosGlennos** [comment]
But then we will lose esthetic value. If I register my account LosGlennos I want it to look like that when the website greets me.

"Welcome back, losglennos" isn't as nice..

**Comment by alveflo** [comment]
That's right... Is it possible to query with case insensitivity on mongodb? If not, I guess the only option would be to expand the database!

**Comment by LosGlennos** [comment]
http://stackoverflow.com/questions/7101703/how-do-i-make-case-insensitive-queries-on-mongodb

You make the call.

**Comment by alveflo** [comment]
Hmm... Go ahead and create a _lower-field. If we come up with a better solution later on, it's easy to fix.

Proposed solution approved! :+1:
